test/simulator_tests/simple_fit_subclone_simulator_test.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+from typing import Generator
+import unittest
+
+from cassiopeia.simulator import SimpleFitSubcloneSimulator
+
+
+class TestSimpleFitSubcloneSimulator(unittest.TestCase):
+    def test_deterministic(self):
+        r"""
+        Small test that can be drawn by hand.
+        Checks that the generated phylogeny is correct.
+        """
+        tree = SimpleFitSubcloneSimulator(
+            branch_length_neutral=1,
+            branch_length_fit=0.5,
+            experiment_duration=1.9,
+            generations_until_fit_subclone=1,
+        ).simulate_tree()
+        self.assertListEqual(
+            tree.nodes,
+            ["0_neutral", "1_neutral", "2_fit", "3_neutral", "4_fit", "5_fit"],
+        )
+        self.assertListEqual(
+            tree.edges,
+            [
+                ("0_neutral", "1_neutral"),
+                ("1_neutral", "2_fit"),
+                ("1_neutral", "3_neutral"),
+                ("2_fit", "4_fit"),
+                ("2_fit", "5_fit"),
+            ],
+        )
+        self.assertDictEqual(
+            tree.get_times(),
+            {
+                "0_neutral": 0.0,
+                "1_neutral": 1.0,
+                "2_fit": 1.5,
+                "3_neutral": 1.9,
+                "4_fit": 1.9,
+                "5_fit": 1.9,
+            },
+        )
+
+    def test_stochastic(self):
+        r"""
+        We test the functionality that allows providing a generator of branch
+        lengths. Because the test is stochastic, we don't assert anything
+        besides the branch lengths being all different.
+        """
+        np.random.seed(1)
+
+        def branch_length_neutral_generator() -> Generator[float, None, None]:
+            while True:
+                yield np.random.exponential(1.0)
+
+        def branch_length_fit_generator() -> Generator[float, None, None]:
+            while True:
+                yield np.random.exponential(0.5)
+
+        tree = SimpleFitSubcloneSimulator(
+            branch_length_neutral=branch_length_neutral_generator(),
+            branch_length_fit=branch_length_fit_generator(),
+            experiment_duration=4.9,
+            generations_until_fit_subclone=2,
+        ).simulate_tree()
```

Comment:
I think it's a bit odd to expect users to pass in these generators. I understand you want to support both deterministic and random branch lengths, which I agree is nice, but I think there's a more straightforward way to have branch lengths be drawn from a random distribution. Could you have `SimpleFitSubcloneSimulator` take in a function or a constant and turn either of these into an anonymous function or a Generator?  Or have `SimpleFitSubcloneSimulator` take in an anonymous function altogether?